index.js

```diff
@@ -273,11 +280,18 @@ class SideMenu extends Component {
    * @return {React.Component}
    */
   render() {
+    let menu = null;
+
+    /**
+     * If menu is ready to be rendered
+     */
+    if (this.state.shouldRenderMenu) {
+      menu = <View style={styles.menu}>{this.props.menu}</View>;
+    }
+
     return (
       <View style={styles.container} onLayout={this.onLayoutChange.bind(this)}>
-        <View style={styles.menu}>
-          {this.props.menu}
-        </View>
+        {menu}
```

Comment:
Why not inline? Matter of taste, but I am not a huge fan of reassigning variables ;)
E.g.
```
{this.state.shouldRenderMenu && (
   <View style={styles.menu}>{this.props.menu}</View>
)}
```